### PR TITLE
AC-646: Hid subaccount section if subaccount user

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 - Templates hides field for subaccount users on template edit page
-  Snippets hides field for subaccount users on snippet edit page

### How To Test
 - To Add subaccount users:
    - [Use this link](https://github.com/SparkPost/2web2ui/blob/master/docs/feature-flags.md#give-access-to-your-feature) to add the ui option flag "subaccount_reporting"
    - Use an admin account to create a subaccount
    - Use the admin account to invite a subaccount reporting user with "Manage Users" -> "Invite User"
      - Invite email doesn't work on appteam because SSO so I recommend creating an account
 - Check that the fields are absent on subaccount users
 - Check that the fields are present on non subaccount users
 - Check that form submission works for users with edit permissions

### To Do
- [ ] Address feedback

### CODE REVIEW

File: src/pages/snippets/tests/EditPage.test.js [hunk 1]
```diff
@@ -34,7 +36,12 @@ describe('EditPage', () => {
 
   it('renders without subaccount section when account has no subaccounts', () => {
     const wrapper = subject({ hasSubaccounts: false });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find(SubaccountSection)).not.toExist();
+  });
+
+  it('renders without subaccount section when user is a subaccount user', () => {
```
[the remark this answers is not in the document]
Comment (PatrickSison) [reply src/pages/snippets/tests/EditPage.test.js]: Yup, updated. Initially I had left in the initial render snapshot, but went ahead and added explicit assertion.

[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/snippets/tests/EditPage.test.js]: I see why I didn't notice it in snapshot. Looks like it's rendering as `Component` which isn't very meaningful. how about setting a `displayName` property to `SubaccountSection.container` so it has a friendlier name like `<SubaccountSection>`?

```export default withFormName(connect(mapStateToProps, { change })(SubaccountSection));```
to 
```
const connectedComponent = withFormName(connect(mapStateToProps, { change })(SubaccountSection));
connectedComponent.displayName = 'SubaccountSection';
export default connectedComponent;
```

File: src/pages/snippets/tests/EditPage.test.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
```
Comment (rajumsys) [inline src/pages/snippets/tests/EditPage.test.js]: can we convert `renders loading` snapshot to inline assertion? it's a simple improvements

**Comment by rajumsys** [comment]
@PatrickSison Thanks for adding instructions. I could follow them and test the feature. 

I've left two comments but those are optional and up to you but IMO, will be good improvements. Either way :shipit:

**Comment by rajumsys** [comment]
All good, let's ship it tomorrow.

**Pull request merged**